### DISCUSSION TITLE
Kestrel setup

### DISCUSSION THREAD

**Discussion by bjornhellander** [CoreWCF/CoreWCF#1211] [General]
Hi!

I have successfully ported a rather large self-hosted WCF application with lots of services to use CoreWCF instead, but have some questions in case someone has time to offer some insights.

Since we are still using .NET Framework, I used WebHost in ASP.NET 2 to do the setup. I used `UseKestrel(o => o.ListenAnyIP(0))` to configure the hosting, since I didn't get it to work without it. This does work, but I really have no use for that port. I also fear that it could potentially lead to configuration issues in a complicated system with many service processes, but that is not my main concern. In the context of only using Kestrel for hosting CoreWCF services, do I need to have Kestrel open this (for me) unnecessary port?

**Comment by kdrapel** [comment]
Hi, I did not manage to have Kestrel without any listening. Which somehow makes sense because that's the root job of Kestrel to listen to something at some point. However I looked into the internal to see how the ListenAnyIP(0) was flowing into the guts of Kestrel and I arrived on a section where it indicates that "The following two lists configure the endpoints that Kestrel should listen to. If both lists are empty, the "urls" config setting (e.g. UseUrls) is used.".

I did not investigate reeally further but the alternative could be to have this .UseStartup(...).UseUrls("xxxx") - I don't know if you can achieve that without an url containing a port. I made a quick test with what I have on my side, and a call to UseUrls("0.0.0.0") without the ListenAnyIP part and it does not work...

![image](https://github.com/CoreWCF/CoreWCF/assets/12449790/874832a8-4f5f-4518-9742-fdf32a6c6fe4)
